hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointImpl.java

```diff
@@ -236,4 +255,36 @@ public String toString() {
                 + ", latest clientAttributes=" + getClientAttributes()
                 + '}';
     }
+
+    @Override
+    public void provideDynamicMetrics(MetricDescriptor descriptor, MetricsCollectionContext context) {
+        ClientStatistics clientStatistics = statsRef.get();
+        if (clientStatistics != null && clientStatistics.metricsBlob() != null) {
+            long timestamp = clientStatistics.timestamp();
+            byte[] metricsBlob = clientStatistics.metricsBlob();
+            MetricConsumer consumer = new MetricConsumer() {
+                @Override
+                public void consumeLong(MetricDescriptor descriptor, long value) {
+                    context.collect(enhanceDescriptor(descriptor, timestamp), value);
+                }
+
+                @Override
+                public void consumeDouble(MetricDescriptor descriptor, double value) {
+                    context.collect(enhanceDescriptor(descriptor, timestamp), value);
+                }
+
+                private MetricDescriptor enhanceDescriptor(MetricDescriptor descriptor, long timestamp) {
+                    return descriptor
+                        // we exclude all metric targets here besides MANAGEMENT_CENTER
+                        // since we want to send the client-side metrics only to MC
+                        .withExcludedTargets(asList(MetricTarget.values()))
```

Comment:
`MetricTarget.values()` creates a new array instance on each invocation (as it's standard enum behavior). We could cache this array as a static field in this anonymous class or do something similar to avoid paying for this.

Reply:
Added a static array and list for this purpose to `MetricTarget` and using that instead of `values()`.